frontend/src/components/editor/output/MarimoTracebackOutput.tsx

```diff
@@ -94,3 +102,85 @@ function lastLine(text: string): string {
   const lines = el.textContent?.split("\n").filter(Boolean);
   return lines?.at(-1) || "";
 }
+
+const matchesSelector = (domNode: Element, selector: string) => {
+  const [tagName, ...classes] = selector.split(".");
+  const classList = (domNode.attribs.class || "").split(" ");
+  return (
+    domNode.tagName === tagName &&
+    classes.every((cls) => classList.includes(cls))
+  );
+};
+
+const elementContainsMarimoCellFile = (domNode: Element) => {
+  return (
+    domNode &&
+    matchesSelector(domNode, "span.nb") &&
+    domNode.firstChild instanceof Text &&
+    domNode.firstChild.nodeValue?.includes("__marimo__")
+  );
+};
+
+const replaceTracebackFilenames = (domNode: DOMNode) => {
```

Comment:
if not too much trouble, i think we should add a unit test for this function

Reply:
(also happy to do any of these unit tests if you want)

Reply:
Added some tests, but maybe not as targeted as they should be, feel free to add more if some edges cases might be lost